### DISCUSSION TITLE
Snap package

### DISCUSSION THREAD

**Discussion by faveoled** [chidiwilliams/buzz#484] [Show and tell]
Hey. I've packaged Buzz into a Snap package (mostly for myself). Should work on multiple Linux distros. Check it out:

[![Get it from the Snap Store](https://snapcraft.io/static/images/badges/en/snap-store-black.svg)](https://snapcraft.io/buzz-fo)

**Comment by chidiwilliams** [comment]
@faveoled Could you please share your snapcraft.yml file?

**Reply by chidiwilliams** [reply]
Nvm, found it here, thanks: https://github.com/faveoled/buzz-fo-snap/tree/main. I'll try to add it to the repo and build job.

**Reply by faveoled** [reply]
@chidiwilliams snaps are usually installed through the software store - you might consider registering on snapcraft.io yourself

**Comment by hardbrocklife** [comment]
With your snap I get this error when buzz attempts to download the models:

Download failed: [Errno 18] Invalid cross-device link: '/tmp/tmp2guykkl4' -> '/home/hardbrocklife/snap/buzz-fo/common/.cache/base.pt'

Is this some kind of permission error? When I did some research on it I all I found is that its an error thrown by python when copying files.

Any clue on wjat I should do or look for to help resolve the issue?

Specs:
Operating System: Manjaro Linux
KDE Plasma Version: 5.27.5
KDE Frameworks Version: 5.106.0
Qt Version: 5.15.9
Kernel Version: 6.1.31-2-MANJARO (64-bit)
Graphics Platform: X11
Processors: 24 × AMD Ryzen 9 3900X 12-Core Processor
Memory: 62.7 GiB of RAM
Graphics Processor: NVIDIA GeForce RTX 3080/PCIe/SSE2

**Reply by faveoled** [reply]
It's a bug #454 (not related to snap). Should be fixed in the next release. For me downloading the *.pt file manually works.

**Comment by hardbrocklife** [comment]
Hmm. My root partition where tmp is located is on a different partition from my home folder. I wonder if thats why. 

I tried manually copying renaming the archive to base.pt and extracting it and renaming the folder base.pt, just to see what would happen and no dice.

**Reply by faveoled** [reply]
You shouldn't extract it. Should be placed like a file e.g. `/home/user/snap/buzz-fo/common/.cache/tiny.pt`

**Reply by hardbrocklife** [reply]
I didnt explain myself well earlier. That is one of the things I had tried. Just copying it straight to the folder and renaming appropriately.

When I did that I get "Failed (Unknown Error)" within the Buzz application when I try to transcribe a file.

While retrying everything from start to finish to make sure I did everything properly, I stumbled upon the buzz logs.txt and seen this:

"[2023-06-11 14:23:28,229] transcriber.read_line:586 DEBUG -> whisper (stderr): torch/cuda/__init__.py:146: UserWarning: 
NVIDIA GeForce RTX 3080 with CUDA capability sm_86 is not compatible with the current PyTorch installation.
The current PyTorch install supports CUDA capabilities sm_37 sm_50 sm_60 sm_70.
If you want to use the NVIDIA GeForce RTX 3080 GPU with PyTorch, please check the instructions at https://pytorch.org/get-started/locally/"

Im going to assume Buzz itself is working properly, but I have a dependency issue. I visited the link suggested, input my system info, and updated everything with pip command the pytorch website suggested.

Researching found this thread:
[NVIDIA GeForce RTX 3090 with CUDA capability sm_86 is not compatible with the current PyTorch installation](https://discuss.pytorch.org/t/nvidia-geforce-rtx-3090-with-cuda-capability-sm-86-is-not-compatible-with-the-current-pytorch-installation/141940)

I check the current versions:

`~ python3 -m pip list | grep torch   
torch                    2.0.1
torchaudio               2.0.2
torchvision              0.15.2

and checked cuda:

` ~  nvidia-smi                                                                                                                                                 
Sun Jun 11 15:10:11 2023 
| NVIDIA-SMI 530.41.03              Driver Version: 530.41.03    CUDA Version: 12.1     |


nvcc - version returns nothing. 

Based on the research I have found most people uninstalled the pip version and reinstalled with the conda and it solved the problem as it looks like installing with conda installs all dependencies. I think I understand now that a CUDA version and CUDA runtime are not synonymous, and that CUDA runtime is the dependency I am missing?

**Reply by faveoled** [reply]
I think it uses torch integrated into snap which is older. Now it could be described as a snap issue which I'm not currently sure how to fix properly. Thanks for the report, I'll try to look into it.

**Reply by hardbrocklife** [reply]
No worries. The linux zip from the repo does the same thing, even after all the changes I made to try to update torch and cuda to the most recent releases. 

`2023-06-12 11:21:00,133] transcriber.run:732 DEBUG -> Starting next transcription task
[2023-06-12 11:21:00,162] transcriber.transcribe:436 DEBUG -> Starting whisper file transcription, task = FileTranscriptionTask(file_path='/run/media/hardbrocklife/linuxstor/InternetVideo/ThingsBoard I Mersino (2023-06-09 1602 GMT3).mp4', transcription_options=TranscriptionOptions(language='en', task=<Task.TRANSCRIBE: 'transcribe'>, model=TranscriptionModel(model_type=<ModelType.WHISPER: 'Whisper'>, whisper_model_size=<WhisperModelSize.BASE: 'base'>, hugging_face_model_id=None), word_level_timings=False, temperature=(0.0, 0.2, 0.4, 0.6, 0.8, 1.0), initial_prompt='', openai_access_token=''), file_transcription_options=FileTranscriptionOptions(file_paths=['/run/media/hardbrocklife/linuxstor/InternetVideo/ThingsBoard I Mersino (2023-06-09 1602 GMT3).mp4'], output_formats=set()), model_path='/home/hardbrocklife/.cache/whisper/base.pt', id=126514, segments=[], status=<Status.QUEUED: 'queued'>, error=None)
**[2023-06-12 11:21:01,094] transcriber.read_line:586 DEBUG -> whisper (stderr): torch/cuda/__init__.py:146: UserWarning: 
NVIDIA GeForce RTX 3080 with CUDA capability sm_86 is not compatible with the current PyTorch installation.
The current PyTorch install supports CUDA capabilities sm_37 sm_50 sm_60 sm_70.
If you want to use the NVIDIA GeForce RTX 3080 GPU with PyTorch, please check the instructions at https://pytorch.org/get-started/locally/**

  warnings.warn(incompatible_device_warn.format(device_name, capability, " ".join(arch_list), device_name))
[2023-06-12 11:21:10,597] transcriber.transcribe:458 DEBUG -> whisper process completed with code = 1, time taken = 0:00:10.435431, number of segments = 0
[2023-06-12 11:21:10,597] transcriber.run:250 ERROR -> 
Traceback (most recent call last):
  File "buzz/transcriber.py", line 247, in run
  File "buzz/transcriber.py", line 463, in transcribe
Exception: Unknown error
[2023-06-12 11:21:10,598] transcriber.run:716 DEBUG -> Waiting for next transcription task`

Thanks for trying to help regardless.
